infra/modules/auth-github-actions/variables.tf

```diff
@@ -8,14 +8,8 @@ variable "github_repository" {
   description = "The GitHub repository in 'org/repo' format to provide access to AWS account resources. Example: navapbc/template-infra"
 }
 
-variable "github_branch" {
-  type        = string
-  description = "The git refs of the GitHub repo to provide access to AWS account resources. Defaults to '*' which allows all branches. Example: refs/heads/main"
-  default     = "*"
-}
-
 variable "iam_role_policy_arns" {
   type        = list(string)
   description = "List of IAM policy ARNs to attach to the GitHub Actions IAM role. Defaults to Developer power user access role."
-  default     = ["arn:aws:iam::aws:policy/PowerUserAccess"]
+  default     = ["arn:aws:iam::aws:policy/AdministratorAccess"]
```

Comment:
@kyeah maybe for future discussion, i realized that the power user access doesn't give ability to create IAM roles, so for now I defaulted the github actions role to have administrator access. What are your thoughts on this?

It seems like the IAM roles for things like the application service should be created by the developer / CD, since you need to create the role per environment, but on the other hand needing administrator access might be an issue.

Reply:
yeah "who should have power to create IAM roles" is kind of a pain. I agree that creating IAM roles along with the application service is the most sensical.

In more strict security environments, I think we'd want to support a Github Actions role here that can create other IAM roles, but within limits (using Permission Boundaries.)

I haven't used that before but I think that's the intention. In this example from AWS, there's an administrative step to create the permission boundary, and then the "employee" (Github Actions in this case) can create new roles within the permission boundary:

https://aws.amazon.com/blogs/security/delegate-permission-management-to-developers-using-iam-permissions-boundaries/

That doesn't address the potential need for developers to be able to run terraform locally though (outside of CI/CD). Ideally both the Github Actions and developer roles are created with the same permission boundaries, butwWe've seen on PFML that some orgs want developer IAM policies to be managed centrally instead of being created in Terraform.

Reply:
Yeah I think that's a good idea. Any idea if the permissions boundary can be set with terraform?

Created a ticket for this for now https://github.com/navapbc/template-infra/issues/194

Reply:
Late reply but yeah I think it's possible via terraform. IAM roles have a [permissions_boundary](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role#permissions_boundary) field which would reference a IAM policy document defining the boundaries.

Reply:
Awesome thanks for finding that